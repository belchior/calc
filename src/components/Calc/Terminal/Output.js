import React, {Component} from 'react';

class Output extends Component {
  componentDidUpdate() {
    this.outputElem.scrollTop = this.outputElem.scrollHeight;
  }

  render() {
    const results = this.props.results || [];
    return (
      <div className="output" ref={(output) => this.outputElem = output}>
        {results.map((item, index) => <output className="result" key={index}>{item}</output>)}
      </div>
    );
  }
}

export default Output;
