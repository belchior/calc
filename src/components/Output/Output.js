import React, {Component} from 'react';

export class Output extends Component {
  componentDidUpdate() {
    this.outputElem.scrollTop = this.outputElem.scrollHeight;
  }

  render() {
    const results = this.props.results || [];
    return (
      <div className="output" ref={(output) => this.outputElem = output}>
        {results.map((item, index) => {
          return <output className="result" onClick={this.props.onClick} key={index}>{item}</output>;
        })}
      </div>
    );
  }
}

export default Output;
