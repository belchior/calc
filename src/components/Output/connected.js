import { connect } from 'react-redux';
import Output from './Output';

export const mapStateToProps = (state) => {
  return state.terminal;
};

export const mapDispatchToProps = (dispatch) => {
  return {
    onClick: (e) => {
      dispatch({ type: 'NUMBER', calc: 'termsinal', payload: e.target.textContent });
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Output);
