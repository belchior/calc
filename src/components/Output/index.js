import Output, { mapStateToProps, mapDispatchToProps } from './connected';

export { mapStateToProps, mapDispatchToProps, };
export default Output;
