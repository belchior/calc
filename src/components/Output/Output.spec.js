import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Output from './Output';


const setup = (props = {}) => {
  const requiredProps = {
    onClick: () => {},
    ...props,
  };
  return shallow(<Output {...requiredProps} />);
};


describe('Output', () => {
  it('should render Output with 0 items without crashing', () => {
    const component = () => setup();
    expect(component).not.toThrow();
  });

  it('should render Output with 2 items without crashing', () => {
    const props = { results: [ '1.2', '3.4' ] };
    const wrapper = setup(props);
    const custom = toJson(wrapper);

    expect(custom).toMatchSnapshot();
  });
});
