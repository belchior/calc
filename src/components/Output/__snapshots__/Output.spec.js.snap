// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Output with 2 items without crashing 1`] = `
<div
  className="output"
>
  <output
    className="result"
    key="0"
    onClick={[Function]}
  >
    1.2
  </output>
  <output
    className="result"
    key="1"
    onClick={[Function]}
  >
    3.4
  </output>
</div>
`;
